atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -120,10 +112,17 @@ public TransactionScopedCache createTransactionScopedCache(long startTs) {
     }
 
     private synchronized void processCommitUpdate(long startTimestamp) {
-        CommitUpdate commitUpdate = eventCache.getCommitUpdate(startTimestamp);
         Optional<TransactionScopedCache> cache = cacheStore.getCache(StartTimestamp.of(startTimestamp));
+        cache.ifPresent(TransactionScopedCache::close);
```

Comment:
I moved close back here. Reason being that I think we want to ensure the cache is closed before processing commit updates, as that means that the values will not change from this point onwards.

Reply:
Then I would argue it shouldn't be called `close()`? Maybe call it `freeze()` or something? The design of close then call methods is suspicious